dom.go

```diff
@@ -454,6 +489,24 @@ func (tl *TokenList) Set(s []string) {
 	tl.SetString(strings.Join(s, " "))
 }
 
+type attribute struct {
+	key   string
+	value string
+}
+
+func (a attribute) Value() string {
+	return a.value
+}
+
+func (a attribute) Key() string {
+	return a.key
+}
+
+type Attribute interface {
```

Comment:
To partially answer my own second question: Attributes doesn't seem to have a meaningful order

> the Attr nodes' indexes may differ among browsers. To be more specific, attributes is a key/value pair of strings

The documentation for dataset isn't as specific, but since it also returns a map, (and a subset of attributes), it would seem to me that it, too, has no defined order.